pkg/rancher-desktop/store/credentials.ts

```diff
@@ -3,9 +3,33 @@ import { ActionContext, MutationsType } from './ts-helpers';
 
 import type { ServerState } from '@pkg/main/commandServer/httpCommandServer';
 import { ipcRenderer } from '@pkg/utils/ipcRenderer';
+import Latch from '@pkg/utils/latch';
+
+export type Credentials = Omit<ServerState, 'pid'>;
 
 interface CredentialsState {
-  credentials: Omit<ServerState, 'pid'>;
+  credentials: Credentials;
+}
+
+const hasCredentials = Latch();
+
+export async function fetchAPI(api: string, rootState: any, init?: RequestInit) {
```

Comment:
Right, I don't like having this attached to the credentials store, but this might be the best place for it for now.. I think that we leave it as is for now and give this more thought if we find ourselves doing more changes like this. 

I'm fine with this approach just as long as we take care to keep its surface area restricted to store to store communication.